angular_components/lib/simple_html/README.md

```diff
@@ -0,0 +1,297 @@
+
+```html
+<simple-html [contents]="someMessage"
+             (trigger)="recordAnalyticsEvent()"></simple-html>
+```
+
+By default, the normal behaviour of the link also occurs, such as navigation. So
+if a user clicks "here" in this message
+
+```html
+Click <a href="//example.com" class="trigger" (trigger)="myTrigger()">here</a>
+```
+
+then they will navigate to example.com _and_ `myTrigger` will be executed.
+
+The trigger definition can also refer to the original Angular
+[`$event`](https://webdev.dartlang.org/angular/guide/template-syntax#event-and-event-handling-statements).
+This can be useful for more advanced features such as:
+
+*   Selectively preventing the default behaviour (such as navigation) by calling
+    [`$event.preventDefault`](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault).
+*   Having multiple `trigger` elements with different behaviour in the same
+    message. Which element was clicked can be identified by reading the
+    [`$event.target`](https://developer.mozilla.org/en-US/docs/Web/API/Event/target)
+    field.
+
+### Limitations
+
+The following is not yet supported:
+
+*   Other elements and attributes that are probably safe but are not yet in
+    demand (such as `blockquote`).
+*   More dangerous cases (such as setting the URL of an iframe) must still
+    be done using `bypassSecurity*` methods. (But preferably don't do this at
+    all.)
+
+## Migration
+
+Migration from an existing `bypassSecurityTrustHtml` solution is generally
+straightforward. Basic steps:
+
+1.  Import the Dart library
+    `package:angular_components/simple_html/simple_html.dart` in your component
+    file.
+1.  Add the `SimpleHtmlComponent` to your
+    [@Component](https://github.com/angular/angular/blob/master/packages/core/src/metadata/directives.ts#L101)'s
+    directive list.
+1.  For each message that previously used `bypassSecurityTrustHtml`, remove that
+    call and use the string directly.
+1.  In your template file, bind the message with either:
+    *   `<simple-html [contents]="msg"></simple-html>` rather than `<span
+        [innerHtml]="msg"></span>` (if only inline elements are required).
+    *   `<simple-html-block [contents]="msg"></simple-html-block>` rather than
+        `<div [innerHtml]="msg"></div>` (if inline and block elements are
+        required).
+1.  Verify that it works! See [*Testing*](#testing).
+1.  Clean up unused dependencies (e.g. remove the `DomSanitizer` if your
+    component no longer needs it).
+1.  If possible, prevent future usage of the unsafe methods by adding a
+    presubmit check.
+
+Remember that `<simple-html>` is an
+[inline-block](https://developer.mozilla.org/en-US/docs/Web/CSS/display) element
+so, if you need other display types, you will need to apply a custom CSS class
+or wrap it inside another element. Likewise `<simple-html-block>` is a
+block-level element and therefore should not be placed inside inline elements.
+
+### Testing {#testing}
```

Comment:
AFAIK, ` {#testing}` is a [Kramdown][] markdown extension only. GH et all will likely process it as heading text, so consider removing it.

Reply:
Yeah, we are using an expanded feature set and this didn't get scraped out.